packages/table/test/table.test.ts

```diff
@@ -0,0 +1,1043 @@
+        // console.log('NOT this test');
+        const visibilityChangedSpy = spy();
+        const virtualItems = makeItemsTwo(50);
+
+        const el = await fixture<Table>(html`
+            <sp-table
+                selects="multiple"
+                .selected=${['1', '47']}
+                style="height: 120px"
+                .items=${virtualItems}
+                .renderItem=${renderItem}
+                scroller?="true"
+                @visibilityChanged=${({
+                    target,
+                }: Event & { target: Table }) => {
+                    visibilityChangedSpy(target);
+                }}
+            >
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+            </sp-table>
+        `);
+        await nextFrame();
+        await nextFrame();
+        await nextFrame();
+
+        expect(el.selected).to.deep.equal(['1', '47']);
+
+        el.scrollToIndex(47);
+
+        await nextFrame();
+        await nextFrame();
+        await elementUpdated(el);
+
+        expect(visibilityChangedSpy.called).to.be.true;
+    });
+
+    it('selects all checkboxes in Virtualized Table when clicking the TableHeadCheckboxCell', async () => {
+        const virtualItems = makeItemsTwo(50);
+
+        const el = await fixture<Table>(html`
+            <sp-table
+                selects="multiple"
+                .selected=${['1', '47']}
+                style="height: 120px"
+                .items=${virtualItems}
+                .renderItem=${renderItem}
+                scroller?="true"
+            >
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+            </sp-table>
+        `);
+        await elementUpdated(el);
+
+        expect(el.selected).to.deep.equal(['1', '47']);
+
+        const tableHeadCheckboxCell = el.querySelector(
+            'sp-table-head sp-table-checkbox-cell'
+        ) as TableCheckboxCell;
+
+        tableHeadCheckboxCell.checkbox.click();
+
+        await elementUpdated(el);
+    });
+
+    it('validates `value` property to make sure it matches the values in `selected`', async () => {
+        const el = await fixture<Table>(
+            virtualizedCustomValue(virtualizedCustomValue.args as Properties)
+        );
+
+        expect(el.selected).to.deep.equal(['applied-47']);
+    });
+
+    it('surfaces [selects="single"] selection', async () => {
+        const el = await fixture<Table>(selectsSingle());
+
+        expect(el.selected, "'Row 1 selected").to.deep.equal(['row1']);
```

Comment:
Is there a test case where we click a row to ensure that it continues to be a single selection, etc.